pandas/_testing.py

```diff
@@ -1176,20 +1177,23 @@ def assert_series_equal(
             raise AssertionError(msg)
     elif is_interval_dtype(left.dtype) or is_interval_dtype(right.dtype):
         assert_interval_array_equal(left.array, right.array)
-    elif is_datetime64tz_dtype(left.dtype):
-        # .values is an ndarray, but ._values is the ExtensionArray.
+    elif is_categorical_dtype(left.dtype) or is_categorical_dtype(right.dtype):
+        _testing.assert_almost_equal(
+            left._values,
+            right._values,
+            check_less_precise=check_less_precise,
+            check_dtype=check_dtype,
+            obj=str(obj),
+        )
+    elif is_extension_array_dtype(left.dtype) or is_extension_array_dtype(right.dtype):
+        assert_extension_array_equal(left._values, right._values)
```

Comment:
assert_extension_array_equal doesn't work for categorical?

Reply:
We have kwargs `check_categorical` and `check_category_ordered` and a categorical-specific check at the end of `assert_series_equal`.  xref #32571 trying to cut down on these